plus/oop/reverse.js

```diff
@@ -1,7 +1,25 @@
 /**
- * It reverses the given object, i.e. return a copy of the object 
+ * It reverses the given object, i.e. return a copy of the object
  * where the keys have become the values and the values the keys
- * 
+ *
  * @param {object} o the object
  * @returns {object} the new object
- */
\ No newline at end of file
+ */
+
+module.exports = function switchPropAndKey(o) {
+  if (typeof o !== "object") {
+    throw new Error("First parameter must be an object.");
+  }
+  let object = {};
+  let tmp;
+  let keys = Object.keys(o);
+  let values = Object.values(o);
+  for (let i = 0; i < keys.length; i++) {
```

Comment:
Your solution is ok, but you can use a simpler solution for this. There is a for ... method in JavaScript and you can play with keys and values in one line. Could you please make it simpler?